ateccx08/ateccx08.go

```diff
@@ -0,0 +1,258 @@
+// Package ateccx08 provides a driver for the ATECCx08 I2C cryptographic co-processor.
+//
+// Datasheet: https://datasheet.octopart.com/ATSAMA5D27-WLSOM1-Microchip-datasheet-149595509.pdf
+package ateccx08 // import "tinygo.org/x/drivers/ateccx08"
+
+import (
+	"errors"
+	"time"
+
+	"tinygo.org/x/drivers"
+)
+
+var (
+	maxCommandTime = (200 + 50) * time.Millisecond
+)
+
+var (
+	ErrWakeup          = errors.New("error on wakeup")
+	ErrInvalidCRCCheck = errors.New("invalid CRC check")
+	ErrLockFailed      = errors.New("locked failed")
+)
+
+type Device struct {
+	bus     drivers.I2C
+	Address uint8
+}
+
+// New returns ATECCx08 device for the provided I2C bus using default address.
+func New(i2c drivers.I2C) *Device {
+	return &Device{
+		bus:     i2c,
+		Address: Address,
+	}
+}
+
+// Configure the ATECCx08 device.
+func (d *Device) Configure() error {
+	return nil
+}
+
+// Connected returns whether ATECCx08 has been found.
+func (d *Device) Connected() bool {
+	if err := d.Wakeup(); err != nil {
+		return false
+	}
+
+	v, err := d.Version()
+	if err != nil {
+		return false
+	}
+
+	return (v == ATECC508 || v == ATECC608)
```

Comment:
this is fine, but the following is slightly simpler:

```suggestion
	return v == ATECC508 || v == ATECC608
```